### PR TITLE
*: make SquashedTable not implement sql.Node

### PR DESCRIPTION
**Pull request by erizocosmico**
Depends on https://github.com/src-d/go-mysql-server/pull/352

### CODE REVIEW

**Comment by erizocosmico** [comment]
Upgraded go-mysql-server, now everything should pass

**Comment by ajnavarro** [comment]
@erizocosmico can you check codecov? appears that squash iterator is not tested anymore.

**Comment by erizocosmico** [comment]
@ajnavarro it is covered, just some `Close` methods are not being called (dunno why tho) and something that's no longer used and I left there.
I'm investigating why close methods are not being called.

**Comment by erizocosmico** [comment]
Turns out it's not because of this, it's that `ResolvedTable` does not close the `sql.RowIter` returned by `PartitionRows`, only the `sql.PartitionIter` 😅

**Comment by erizocosmico** [comment]
Depends on https://github.com/src-d/go-mysql-server/pull/362

**Comment by ajnavarro** [comment]
@erizocosmico now you can update go-mysql-server and rebase!

File: internal/rule/squashjoins_test.go [hunk 1]
```diff
@@ -33,13 +34,17 @@ func TestAnalyzeSquashJoinsExchange(t *testing.T) {
 	require.NoError(err)
 
 	project, ok := result.(*plan.Project)
+	fmt.Printf("%T", result)
```
Comment (ajnavarro) [inline internal/rule/squashjoins_test.go]: leftover?

**Pull request merged**